MapboxDirections/MBRouteLeg.swift

```diff
@@ -7,16 +7,19 @@ import Polyline
  */
 @objc(MBRouteLeg)
 open class RouteLeg: NSObject, NSSecureCoding {
+    
+    typealias SegmentAttributes = [SegmentAttribute:[Any]]
     // MARK: Creating a Leg
     
-    internal init(steps: [RouteStep], json: JSONDictionary, source: Waypoint, destination: Waypoint, profileIdentifier: MBDirectionsProfileIdentifier) {
+    internal init(steps: [RouteStep], json: JSONDictionary, source: Waypoint, destination: Waypoint, profileIdentifier: MBDirectionsProfileIdentifier, segmentAttributes: SegmentAttributes? = nil) {
         self.source = source
         self.destination = destination
         self.profileIdentifier = profileIdentifier
         self.steps = steps
         distance = json["distance"] as! Double
         expectedTravelTime = json["duration"] as! Double
         self.name = json["summary"] as! String
+        self.segmentAttributes = segmentAttributes
```

Comment:
There should also be two separate properties in RouteOptions, even if they end up being combined into a single `annotations` parameter. It’s unfortunate that the API conflates these two concepts, but there’s no reason we need to extend the confusion to the client library.